File: alligator/src/main/java/me/aartikov/alligator/commands/SwitchToCommand.java
package me.aartikov.alligator.commands;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import me.aartikov.alligator.NavigationContext;
import me.aartikov.alligator.Screen;
import me.aartikov.alligator.animations.AnimationData;
import me.aartikov.alligator.exceptions.MissingScreenSwitcherException;
import me.aartikov.alligator.exceptions.NavigationException;
import me.aartikov.alligator.screenswitchers.ScreenSwitcher;


/**
 * Command implementation for {@code switchTo} method of {@link me.aartikov.alligator.AndroidNavigator}.
 */
public class SwitchToCommand implements Command {
	private Screen mScreen;
	@Nullable
	private AnimationData mAnimationData;

	public SwitchToCommand(@NonNull Screen screen, @Nullable AnimationData animationData) {
		mScreen = screen;
		mAnimationData = animationData;
	}

	@Override
	public boolean execute(@NonNull NavigationContext navigationContext) throws NavigationException {
		ScreenSwitcher screenSwitcher = navigationContext.getScreenSwitcher();
		if (screenSwitcher == null) {
			throw new MissingScreenSwitcherException("ScreenSwitcher is not set.");
		}

		screenSwitcher.switchTo(mScreen, navigationContext.getScreenSwitchingListener(), mAnimationData);
		return true;
	}
}

File: alligator/src/main/java/me/aartikov/alligator/commands/SwitchToCommand.kt
package me.aartikov.alligator.commands

import me.aartikov.alligator.NavigationContext
import me.aartikov.alligator.Screen
import me.aartikov.alligator.animations.AnimationData
import me.aartikov.alligator.exceptions.MissingScreenSwitcherException
import me.aartikov.alligator.exceptions.NavigationException

/**
 * Command implementation for `switchTo` method of [me.aartikov.alligator.AndroidNavigator].
 */
class SwitchToCommand(
    private val mScreen: Screen,
    private val mAnimationData: AnimationData?
) : Command {

    @Throws(NavigationException::class)
    override fun execute(navigationContext: NavigationContext): Boolean {
        val screenSwitcher = navigationContext.screenSwitcher
            ?: throw MissingScreenSwitcherException("ScreenSwitcher is not set.")
        screenSwitcher.switchTo(mScreen, navigationContext.screenSwitchingListener, mAnimationData)
        return true
    }
}